src/v/iceberg/manifest_io.cc

```diff
@@ -55,4 +67,20 @@ manifest_io::upload_manifest_list(
       });
 }
 
+ss::sstring manifest_io::to_uri(const std::filesystem::path& p) const {
+    return fmt::format("{}{}", uri_base(), p.native());
```

Comment:
any concern about std::fs::path::native and if it is always going to format the path the way you want?

Reply:
I don't think so, the logic used to construct them so far is pretty simple (see get_manifest_path and get_manifest_list_path in merge_append_action.cc). Happy to address this if it grows to be more complicated though

Reply:
seems fine to me